hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/services/AbfsClientTestUtil.java

```diff
@@ -56,4 +72,72 @@ public static void setMockAbfsRestOperationForCopyBlobOperation(final AbfsClient
         .when(spiedClient)
         .getCopyBlobOperation(Mockito.any(URL.class), Mockito.anyList());
   }
+
+  public static void setMockAbfsRestOperationForListBlobOperation(
+      final AbfsClient spiedClient) throws Exception {
+    ExponentialRetryPolicy retryPolicy = Mockito.mock(ExponentialRetryPolicy.class);
+    AbfsHttpOperation httpOperation = Mockito.mock(AbfsHttpOperation.class);
+    AbfsRestOperation abfsRestOperation = Mockito.spy(new AbfsRestOperation(
+        AbfsRestOperationType.GetListBlobProperties,
+        spiedClient,
+        HTTP_METHOD_GET,
+        null,
+        new ArrayList<>()
+    ));
+
+    Mockito.doReturn(abfsRestOperation).when(spiedClient).getListBlobOperation(any(), any());
```

Comment:
Better to set behaviors of mocked object in test layer. Reason being, this method in utils can be reused by other tests.

Reply:
Yes, agreed. This was done because the test that require this behavior to set was outside the services package and this behavior can be set only inside services package.

Moreover, this method is to specifically set mocking for listBlob Rest operation and any other test that want to do so, can use this method.

Please let me know your thoughts on this.

Reply:
Has been taken. Thanks!